### PR TITLE
feat: show all transport modes for a stop place

### PR DESCRIPTION
**Pull request by adriansberg**
This PR introduces a significant feature enhancement to our application. Specifically, it extends the functionality to show all modes of transport available for a given stop place.

Prior to this update, our application may not have provided comprehensive transport mode information for each stop place, limiting its overall utility for end-users. With the implementation of this feature, we can now present complete and valuable information on all transport modes for any given stop place, thereby significantly improving the user experience.

The code modifications encompassed by this commit include updates to both the display logic and the data handling components of our application. These changes ensure that our system can retrieve, process, and render all transport mode data associated with each stop place.

In addition to the major feature enhancement, other modifications were made to improve the overall code quality and maintainability. Whilst these changes may not result in noticeable functional differences, they play a crucial role in ensuring that our codebase remains robust, clean and easy to maintain for our development team.

In conclusion, this PR represents a notable improvement to the application. By enhancing the information accuracy provided to the users and improving the underlying code quality, it makes a significant contribution towards our ongoing efforts to deliver an outstanding user experience.

<details><summary>Screenshots</summary>
<p>

2023-11-30 at 18 43 25

</p>
</details> 

Closes https://github.com/AtB-AS/kundevendt/issues/15784

### CODE REVIEW

File: src/components/venue-icon/index.tsx [hunk 1]
```diff
@@ -1,17 +1,26 @@
 import { ComponentText, useTranslation } from '@atb/translations';
 import { MonoIcon, MonoIconProps } from '@atb/components/icon';
+import { TransportModeType } from '@atb/modules/transport-mode';
 
-export type VenuIconProps = {
-  category: FeatureCategory[];
+export type VenueIconProps = {
+  categories?: FeatureCategory[];
```
Comment (adriansberg) [inline src/components/venue-icon/index.tsx:6]: This feels clunky. Initially wanted one property that could be either `FeatureCategory[]` or `TransportModeType[]`, but creating guards to check which type it is was a little comprehensive. Also didn't want to create another mapping function like `mapLocationCategoryToVenueType` only the other way around. 

`TransportModeType` and `VenueIconType` overlaps so I thought maybe we could do it like this.

**Review by mortennordseth — APPROVED** [review]
Looks good. I'm not sure about the `nullable` properties, but I guess it makes sense that a quay can have a name but not a public code etc

File: src/page-modules/assistant/server/journey-planner/validators.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export const tripPatternWithDetailsSchema = z.object({
       line: z
         .object({
           name: z.string(),
-          publicCode: z.string(),
+          publicCode: z.string().nullable(),
```
[the remark this answers is not in the document]
Comment (adriansberg) [reply src/page-modules/assistant/server/journey-planner/validators.ts:101]: I think it's mainly flights.

File: src/components/venue-icon/index.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/components/venue-icon/index.tsx:6]: I couldn't see a better approach to this, but @mikaelbr might have a suggestion about this? 🤔

File: src/components/line-chip/index.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {
 export type LineChipProps = {
   transportMode: TransportModeType;
   transportSubmode?: TransportSubmodeType;
-  publicCode: string;
+  publicCode: string | null;
```
[the remark this answers is not in the document]
Comment (adriansberg) [reply src/components/line-chip/index.tsx:13]: `transportSubmode` is set to `optional()` in the schema, while `publicCode` is `nullable()`.

File: src/components/venue-icon/index.tsx (hunk 1 above)
Comment (adriansberg) [reply src/components/venue-icon/index.tsx:6]: Could separate into different components.

Comment (adriansberg) [reply src/components/venue-icon/index.tsx:6]: I see I wrote "creating guards to check which type it is was a little comprehensive", but can't quite remember why. I'll have to check.

Comment (adriansberg) [reply src/components/venue-icon/index.tsx]: Implemented union type and type guard.
